data-prepper-api/src/main/java/org/opensearch/dataprepper/model/codec/OutputCodec.java

```diff
@@ -25,20 +24,20 @@ public interface OutputCodec {
      * Implementors should do initial wrapping according to the implementation
      *
      * @param outputStream outputStream param for wrapping
+     * @param sinkContext {@link SinkContext} object
      * @throws IOException throws IOException when invalid input is received or not able to create wrapping
      */
-    void start(OutputStream outputStream) throws IOException;
+    void start(OutputStream outputStream, SinkContext sinkContext) throws IOException;
```

Comment:
It may make sense to pass a different object here rather than `SinkContext`. For one, the sink really shouldn't need to know anything about routes. For another, this could vary independently of the `SinkContext`.

Perhaps:

```
interface OutputCodecOptions {
  List<String> getIncludeKeys();
  ...
```

Then you can even create an internal adapter to adapt from `SinkContext` to `OutputCodecOptions`. Or just do a mapping.

Reply:
Thanks David.

As the OutputCodec are loaded via plugin factory and may have its own configuration options (via pluginConfiguration), if we name it something like OutputCodecOptions, it seems to me it's some configuration options belongs to the OutputCodec, in such case, the usage will be 
```
- sink
     -  s3
         codec:
              xxxx: 
                     include_keys: [...]
```

For in our case, the usage as below.
```
- sink
     -  s3
         codec:
              xxxx: 
         include_keys: [...]
         ...
```

So I am not sure if OutputCodecOptions is the right way.  Also, another thing I am not sure is whether or not we are going to throw more info to the SinkContext.  So to me, `Context` seems to be more suitable than `Options` here.  

Thoughts?